includes/civicrm.basepage.php

```diff
@@ -348,6 +332,17 @@ public function basepage_handler($wp) {
 
         global $post;
 
+        // Skip if this not the Base Page.
+        if ($basepage->ID !== $post->ID) {
```

Comment:
This here seems to be the key change in behavior: instead of defaulting to basepage, this only goes to basepage mode if you're on the basepage as specified in the CiviCRM settings.  I imagine this is where to add something where the `if` is `FALSE` if there's a URL param specifying to use basepage mode.

Incidentally, I have no idea whether it's common to use another page as your basepage, but I wanted to flag the following change in behavior.  The examples below all assume that the default page `civicrm` exists and is the basepage set in CiviCRM.  Also, there is another page `other` that exists.

### Before (master)

| URL | Rendered |
| --- | -------- |
| http://example.org/civicrm/?civiwp=CiviCRM&q=civicrm/contribute/transact&reset=1&id=1 | CiviCRM Contribution page ID 1, template of `civicrm`, no page content |
| http://example.org/other/?civiwp=CiviCRM&q=civicrm/contribute/transact&reset=1&id=1 | CiviCRM Contribution page ID 1, template of `other`, no page content |

### After (with this PR)

| URL | Rendered |
| --- | -------- |
| http://example.org/civicrm/?civiwp=CiviCRM&q=civicrm/contribute/transact&reset=1&id=1 | CiviCRM Contribution page ID 1, template of `civicrm`, no page content |
| http://example.org/other/?civiwp=CiviCRM&q=civicrm/contribute/transact&reset=1&id=1 | The normal page content and template on `other`, with nothing from CiviCRM |

The only use case I've encountered of this is a site that had a special template for the conference section of their website and they wanted the registration form to have that look.  Of course, they could use a shortcode for this, instead.  Unlike the other concern about content on a shortcode page, I think this could be easily resolved with an upgrade message.

However, it might be good to run through some of the scenarios @christianwach and @kcristiano were thinking of in #17698 where the recorded basepage may not exist neatly in WP multisite.

Reply:
@agh1 Thanks for the analysis. 

* Yes, you're right; this PR does prevent rendering of CiviCRM content in the `/other` URL scenario (i.e. not on the Base Page) unless embedded via a Shortcode, of course. Personally, I think this kind of URL *should not* function, though I appreciate that people may be using it because it does.
* For the Base Page URL... I'd imagine that most Base Page requests are made using Clean URLs these days as it's been the default for quite some time. But as you point out, nothing changes for these Base Page URLs either way.
* I can't at the moment think of any impact when a Base Page doesn't exist - usually that means either (a) CiviCRM is incorrectly installed and it should have a Base Page or (b) it has no need for front-end content or (c) it's multisite and there's an appropriately configured install that doesn't need a Base Page. Happy to be proved wrong :-)

> `civibasepage=1`

What's your thinking here? How does this help?

Reply:
@agh1 Does 659b8e432463945d68ba0205e694865c56553ea3 solve this for you?

Reply:
@christianwach yep, see comment on the main thread!